### PR TITLE
[skip ci] Remove ceph-disk support

### PR DESCRIPTION
**Pull request by leseb**

### CODE REVIEW

**Comment by leseb** [comment]
This comes on top of: https://github.com/ceph/ceph-ansible/pull/2866.

TODO:

- [ ] remove ceph-disk code

- [ ] update the doc

- [ ] change restart scripts?

- [ ] change handlers?

- [ ] update ceph-validate with the change

- [ ] default osd_scenario to 'lvm' and remove it as a mandatory var

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
@andrewschoen if you could look at the validate part, that'd be nice, thanks.

**Comment by leseb** [comment]
jenkins test luminous-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster_collocation

**Comment by leseb** [comment]
jenkins test luminous-ooo_collocation

**Comment by leseb** [comment]
jenkins test luminous-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster

File: roles/ceph-validate/tasks/check_devices.yml [hunk 1]
```diff
@@ -20,19 +20,20 @@
   register: dedicated_device_parted
   with_items: "{{ dedicated_devices }}"
   when:
-    - osd_scenario == 'non-collocated'
+    - dedicated_devices|default([]) | length > 0
```
Comment (andrewschoen) [inline roles/ceph-validate/tasks/check_devices.yml]: Is ``dedicated_devices`` still needed?

File: roles/ceph-osd/defaults/main.yml [hunk 2]
```diff
@@ -53,110 +53,13 @@ osd_auto_discovery: false
...
-# Example of what you will get:
-# [root@ceph-osd0 ~]# blkid /dev/sda*
-# /dev/sda: PTTYPE="gpt"
-# /dev/sda1: UUID="9c43e346-dd6e-431f-92d8-cbed4ccb25f6" TYPE="xfs" PARTLABEL="ceph data" PARTUUID="749c71c9-ed8f-4930-82a7-a48a3bcdb1c7"
-# /dev/sda2: PARTLABEL="ceph block" PARTUUID="e6ca3e1d-4702-4569-abfa-e285de328e9d"
-#
-# Note: This scenario uses the ceph-disk tool to provision OSDs
-
-osd_scenario: dummy
+osd_scenario: lvm
```
Comment (andrewschoen) [inline roles/ceph-osd/defaults/main.yml]: Do we need ``osd_scenario`` anymore?

File: plugins/actions/validate.py [hunk 3]
```diff
@@ -213,15 +226,6 @@ def validate_rados_options(value):
 osd_options = (
     (optional("dmcrypt"), types.boolean),
     (optional("osd_auto_discovery"), types.boolean),
```
Comment (andrewschoen) [inline plugins/actions/validate.py]: Are we still supporting ``osd_auto_discovery`` when ``ceph-disk`` is removed?

File: tox.ini [hunk 4]
```diff
@@ -247,13 +232,19 @@ commands=
   vagrant up --no-provision {posargs:--provider=virtualbox}
   bash {toxinidir}/tests/scripts/generate_ssh_config.sh {changedir}
 
+  centos7_cluster: ansible-playbook -vv -i {changedir}/hosts {toxinidir}/tests/functional/lvm_setup.yml
```
Comment (andrewschoen) [inline tox.ini]: Is there a reason why would wouldn't always run ``lvm_setup.yml`` now?

**Comment by guits** [comment]
jenkins test luminous-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-docker_cluster_collocation

**Comment by guits** [comment]
jenkins test luminous-ooo_collocation

File: plugins/actions/validate.py (hunk 3 above)
Comment (leseb) [reply plugins/actions/validate.py]: nop!

File: tox.ini (hunk 4 above)
Comment (leseb) [reply tox.ini]: There is not, let me clean that up

File: roles/ceph-validate/tasks/check_devices.yml (hunk 1 above)
Comment (leseb) [reply roles/ceph-validate/tasks/check_devices.yml]: Can't we use it anymore with lvm?

**Comment by leseb** [comment]
jenkins test luminous-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster_collocation

**Comment by leseb** [comment]
jenkins test luminous-ooo_collocation

File: plugins/actions/validate.py (hunk 3 above)
Comment (leseb) [reply plugins/actions/validate.py]: Actually, this is a regression, can't we use `osd_auto_discovery` with batch or something?

Comment (andrewschoen) [reply plugins/actions/validate.py]: There is currently no support for that, batch uses ``devices`` directly. Is ``osd_auto_discovery`` still something we'd like to support?

File: roles/ceph-validate/tasks/check_devices.yml (hunk 1 above)
Comment (andrewschoen) [reply roles/ceph-validate/tasks/check_devices.yml]: the lvm scenarios only use ``devices`` or ``lvm_volumes``. I'm not sure what we'd use it for right now.

Comment (leseb) [reply roles/ceph-validate/tasks/check_devices.yml]: hum, so what if I want to use a raw block device for wal/db, is that doable?

File: plugins/actions/validate.py (hunk 3 above)
Comment (leseb) [reply plugins/actions/validate.py]: I suspect we have users using it, so I'm afraid to see their experience degraded. Even though there is no support for it anymore...

File: roles/ceph-validate/tasks/check_devices.yml (hunk 1 above)
Comment (andrewschoen) [reply roles/ceph-validate/tasks/check_devices.yml]: batch decides where to best put block.db depending on the devices given. At this point batch doesn't do anything with wal I don't believe.

**Comment by leseb** [comment]
jenkins test luminous-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster_collocation

**Comment by guits** [comment]
this PR needs to be resurrected

**Comment by dsavineau** [comment]
this has been done in https://github.com/ceph/ceph-ansible/pull/3837

**Pull request closed without merging**